src/createStore.js

```diff
@@ -196,9 +194,9 @@ export default function createStore(reducer, preloadedState, enhancer) {
       isDispatching = false
     }
 
-    var listeners = currentListeners = nextListeners
-    for (var i = 0; i < listeners.length; i++) {
-      listeners[i]()
+    listeners = currentListeners.slice()
```

Comment:
We used to do this, but it was removed for performance and memory reasons in c031c0a. You're creating multiple duplicate arrays just to avoid race conditions when nesting your dispatches. The simple for loop from before is easier to read and faster.